File: tests/FixpageTest.php
<?php

use PHPUnit\Framework\TestCase;
use function WpRefs\WprefText\fix_page;

class FixpageTest extends TestCase
{
    private function assertEqualCompare(string $expected, string $input, string $result)
    {
        if ($result === $input && $result !== $expected) {
            $this->fail("No changes were made! The function returned the input unchanged:\n$result");
        } else {
            $this->assertEquals($expected, $result, "Unexpected result:\n$result");
        }
    }
    private function fix_page_wrap(string $text, string $lang)
    {
        return fix_page($text, "", true, true, false, $lang, "", 0);
    }

    public function testPart1()
    {
        $input = '[[Category:Translated from MDWiki]] ռետինոիդներ։ <ref name="NORD2006" /><ref name="Gli2017" />';
        // ---
        $expected = '[[Category:Translated from MDWiki]] ռետինոիդներ<ref name="NORD2006" /><ref name="Gli2017" />։';
        // ---
        $this->assertEqualCompare($expected, $input, $this->fix_page_wrap($input, 'hy'));
    }
}

File: tests/FixpageTest.py
from __future__ import annotations

from typing import Callable

import pytest

from src import md_cat
from src.index import fix_page
from tests.conftest import assert_equal_compare


@pytest.fixture(autouse=True)
def stub_mdwiki_category(monkeypatch: pytest.MonkeyPatch) -> Callable[[], None]:
    original = md_cat.get_cats
    if hasattr(original, "cache_clear"):
        original.cache_clear()

    def _get_cats() -> dict:
        return {"hywik": {"title": "Category:Translated from MDWiki"}}

    monkeypatch.setattr(md_cat, "get_cats", _get_cats)
    return _get_cats


class FixpageTest:
    @staticmethod
    def fix_page_wrap(text: str, lang: str) -> str:
        return fix_page(text, "", True, True, False, lang, "", 0)

    def testPart1(self) -> None:
        input_text = (
            "[[Category:Translated from MDWiki]] ռետինոիդներ։ "
            "<ref name=\"NORD2006\" /><ref name=\"Gli2017\" />"
        )
        expected = (
            "[[Category:Translated from MDWiki]] ռետինոիդներ"
            "<ref name=\"NORD2006\" /><ref name=\"Gli2017\" />։"
        )
        assert_equal_compare(expected, input_text, self.fix_page_wrap(input_text, "hy"))
